### PR TITLE
Update to newest version of Uberenv, fixing bugs

### PR DESCRIPTION
**Pull request by adrienbernede**
This PR only updated Uberenv script.

The main fix I wanted is that Uberenv used to ignore a change in the Spack commit section of the `project.json`.
Updating Spack was a pain, or required removing uberenv_libs directory.

### CODE REVIEW

**Comment by adrienbernede** [comment]
LGTM

**Comment by adrienbernede** [comment]
Tested both on PR and Develop workflow. (That’s why it points to the latter in Gitlab CI link).

**Comment by adrienbernede** [comment]
LGTM

File: scripts/uberenv/uberenv.py [hunk 1]
```diff
@@ -224,12 +232,6 @@ def is_darwin():
 def is_windows():
     return "windows" in platform.system().lower()
 
-def is_wsl():
```
Comment (white238) [inline scripts/uberenv/uberenv.py:227]: @jamiebramwell is this still needed for WSL/WSL2 support?

Comment (jamiebramwell) [reply scripts/uberenv/uberenv.py:227]: I don't think so.

File: scripts/uberenv/uberenv.py [hunk 2]
```diff
@@ -272,8 +271,6 @@ def detect_platform(self):
         res = None
         if is_darwin():
             res = "darwin"
-        elif is_wsl():
```
Comment (jamiebramwell) [inline scripts/uberenv/uberenv.py:275]: This is the only case where it is used. We could have wsl users manually specify the spack config (this is what I do for WSL2 anyway), but we should update the docs to reflect this.

**Review by white238 — APPROVED** [review]

File: scripts/uberenv/uberenv.py (hunk 2 above)
Comment (white238) [reply scripts/uberenv/uberenv.py:275]: yea we should just have them specify the directory in the docs.

Comment (jamiebramwell) [reply scripts/uberenv/uberenv.py:275]: 👍 Another PR is incoming to fix the docs.

**Review by jamiebramwell — APPROVED** [review]

**Pull request merged**